ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml

```diff
@@ -131,6 +131,13 @@ tests:
     test:
     - chain: windows-e2e-operator-test-upgrade
     workflow: ipi-vsphere-ovn-hybrid-custom-vxlan-port
+- always_run: false
+  as: azure-e2e-upgrade
+  steps:
+    cluster_profile: azure4
+    test:
+    - chain: windows-e2e-operator-test-upgrade
+    workflow: ipi-azure-ovn-hybrid
```

Comment:
This can replace vsphere-e2e-upgrade rather than add to it

Reply:
vsphere-e2e-upgrade is not functional enough to keep around at the moment.
It can be readded in the future if desired

Reply:
I feel the removal of the `vsphere-e2e-upgrade`  is out of the scope of this PR. I recommend opening a separate PR for it.

Reply:
The ask was to move the upgrade job from vsphere to azure for master.
If anything the creation of the step registry steps, while a beneficial change, would be out of scope :wink:

Reply:
Please just reframe the PR to `Move WMCO bundle upgrade testing to azure for master branch`

Reply:
The takeaway I captured from the design meeting was to add a new job targeting another platform as stated in the Jira [comment](https://issues.redhat.com/browse/OCPBUGS-8996?focusedId=23372457&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-23372457) and in  [Slack](https://redhat-internal.slack.com/archives/CM4ERHBJS/p1698691962243469?thread_ts=1697482639.643899&cid=CM4ERHBJS).

I see your point on reframing the PR to just update the target platform in `vsphere-e2e-upgrade` job  to another that does not present the docker limit issue, for example, `azure-e2e-upgrade`.  Will make the adjustments.

Reply:
> Please just reframe the PR to `Move WMCO bundle upgrade testing to azure for master branch`

Addressed in https://github.com/openshift/release/commit/13aeb5f748c93e1cef6bd6fad633e710d6bd92cd

@sebsoto  PTAL.